_vector-search/ai-search/hybrid-search/inner-hits.md

```diff
@@ -0,0 +1,927 @@
+        "number_of_replicas": 0
+    },
+    "mappings": {
+        "properties": {
+            "user": {
+                "type": "nested",
+                "properties": {
+                    "name": {
+                        "type": "text"
+                    },
+                    "age": {
+                        "type": "integer"
+                    }
+                }
+            },
+            "location": {
+                "type": "nested",
+                "properties": {
+                    "city": {
+                        "type": "text"
+                    },
+                    "state": {
+                        "type": "text"
+                    }
+                }
+            }
+        }
+    }
+}
+```
+{% include copy-curl.html %}
+
+### Step 2: Create search pipeline with normalization-processor
+Configure a search pipeline with a normalization-processor, use the following request. The normalization technique in the processor is set to min_max, and the combination technique is set to arithmetic_mean.
+```json
+PUT /_search/pipeline/nlp-search-pipeline
+{
+  "description": "Post processor for hybrid search",
+  "phase_results_processors": [
+    {
+      "normalization-processor": {
+        "normalization": {
+          "technique": "min_max"
+        },
+        "combination": {
+          "technique": "arithmetic_mean",
+          "parameters": {}
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Step 3: Ingest the documents into the index
+To ingest documents into the index created in the previous step, send the following requests:
+```json
+PUT /my-nlp-index/_doc/1
+{
+    "user": [
+        {
+            "name": "John Alder",
+            "age": 35
+        },
+        {
+            "name": "Sammy",
+            "age": 34
+        },
+        {
+            "name": "Mike",
+            "age": 32
+        },
+        {
+            "name": "Maples",
+            "age": 30
+        }
+    ],
+    "location":[
+        {
+            "city":"Amsterdam",
+            "state":"Netherlands"
+        },
+        {
+            "city":"Udaipur",
+            "state":"Rajasthan"
+        },
+        {
+            "city":"Naples",
+            "state":"Italy"
+        }
+    ]
+}
+```
+{% include copy-curl.html %}
+```json
+PUT /my-nlp-index/_doc/2
+{
+    "user": [
+        {
+            "name": "John Wick",
+            "age": 46
+        },
+        {
+            "name": "John Snow",
+            "age": 40
+        },
+        {
+            "name": "Sansa Stark",
+            "age": 22
+        },
+        {
+            "name": "Arya Stark",
+            "age": 20
+        }
+    ],
+    "location":[
+        {
+            "city":"Tromso",
+            "state":"Norway"
+        },
+        {
+            "city":"Los Angeles",
+            "state":"California"
+        },
+        {
+            "city":"London",
+            "state":"UK"
+        }
+    ]
+}
```

Comment:
Let's add _bulk ingestion instead of separate doc ingestion to avoid the length of the doc

Reply:
The reason I preferred using this way is because it clearly illustrates the indexing document by mapping to a docId. If I use bulk ingestion, then 2 issues will arise.
1. This json has nested fields which when put after removing spacing in the bulk ingestion request will create issues for user to understand. It will look little messy.
2. If we map it clearly with docId then it becomes little easy to understand in the response. 

Open for your suggestion.

Reply:
Since ingestion is not the focus of the blogpost. Let's avoid more detailing here and use bulk api

Reply:
Actually, nested fields and parent-child can only support inner hits. So indexing will matter here.